internal/parse/parse.go

```diff
@@ -41,6 +41,67 @@ var (
 	errDuplicateDest = errors.New("duplicate mount destination")
 )
 
+// BuildOutputOptions contains parsed field generated by parsing build output string
+type BuildOutputOption struct {
+	Path     string
+	IsDir    bool
+	IsStdout bool
+}
+
+// GetBuildOutput is responsible for parsing custom build output argument.
+// Takes `buildOutput` as string and returns path where output has to be generated returns BuildOutputOptions
+func GetBuildOutput(buildOutput string) (BuildOutputOption, error) {
+	isDir := true
+	isStdout := false
+	typeSelected := false
+	pathSelected := false
+	path := ""
+	if len(buildOutput) == 1 && buildOutput == "-" {
+		// Feature parity with buildkit, output tar to stdout
+		// Read more here: https://docs.docker.com/engine/reference/commandline/build/#custom-build-outputs
+		return BuildOutputOption{"", false, true}, nil
+	}
+	if !strings.Contains(buildOutput, ",") {
+		// expect default --output <dirname>
+		return BuildOutputOption{buildOutput, isDir, isStdout}, nil
+	}
+	tokens := strings.Split(buildOutput, ",")
+	for _, option := range tokens {
+		arr := strings.SplitN(option, "=", 2)
+		if len(arr) != 2 {
+			return BuildOutputOption{}, fmt.Errorf("invalid build output options %q", buildOutput)
```

Comment:
We know exactly why we're rejecting the provided value here, so the error message doesn't need to be that vague.